File: 백준/Silver/10828. 스택/스택.py
import sys

stack = list()
lineCount = int(input())

for i in range(lineCount):
    inputLine = sys.stdin.readline().split()
    order = inputLine[0]
    if order=='push':
        stack.append(int(inputLine[1]))

    if order=='pop':
        if len(stack)==0:
            print(-1)
            continue
        print(stack.pop())

    if order=='size':
        print(len(stack))

    if order=='empty':
        if len(stack)==0:
            print(1)
        else:
            print(0)

    if order=='top':
        if len(stack)==0:
            print(-1)
            continue
        print(stack[len(stack)-1])

File: 백준/Silver/10828. 스택/스택.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int orderCnt = Integer.parseInt(br.readLine());
		List<String> stack = new ArrayList<>();
		
		for (int i = 0; i < orderCnt; i++) {
			String[] inputList = br.readLine().split(" ");
			
			String answer;
			switch(inputList[0]) {
				case "push":
					stack.add(inputList[1]);
					break;
				case "pop":
					System.out.println(stack.isEmpty()?-1:stack.remove(stack.size()-1));
					break;
				case "size":
					System.out.println(stack.size());
					break;
				case "empty":
					System.out.println(stack.isEmpty()?1:0);
					break;
				case "top":
					System.out.println(stack.isEmpty()?-1:stack.get(stack.size()-1));
					break;
			}
		}
	}
}